react/modules/module03/README.md

```diff
@@ -1,68 +1,98 @@
 # Week 3
 
-## React. Hooks
+## React. Tests/Context API
 
-Task should be based on the previous task.
+The task should be based on the previous task.
 
 ### What should be done:
 
-1. Create a separate branch for this task from the previous branch task.
-2. Retain the functionality and rewrite the main page with the hooks. All tests should be working in the end
-3. Retain the functionality and rewrite the form with React Hook Form. All tests should be working in the end
+1. Create a separate branch for this task from the previous task's branch.
+2. Implement custom state management using the Context API.
+   - Utilize the created context to store both the value entered in the Search component and the list of items received from the API;
+   - Ensure that all components that need to access this data use the context.
+3. Add and configure testing libraries: React Testing Library and Jest.
+4. Write tests for the following scenarios, keeping in mind that you should use mocked data instead of real API calls:
+   - Card List:
```

Comment:
May be it worth puting each requirement for tests as a separate on, e.g.:
```
...
3. Add and configure test runner: Jest or Vitest. Add testing library: React Testing Library. You should add tests for the several scenarios keeping in mind that mocked data should be used instead of real API calls.
4. Tests for the Card List:
     - Verify that the component renders the specified number of cards;
     - Check that an appropriate message is displayed if no cards are present.
5.  Tests for the Card component:
...